libbpf-rs/src/perf_buffer.rs

```diff
@@ -87,6 +91,18 @@ impl<'a, 'b> PerfBufferBuilder<'a, 'b> {
             pages,
             sample_cb: self.sample_cb,
             lost_cb: self.lost_cb,
+            wakeup_events: self.wakeup_events,
+        }
+    }
+
+    /// Minimum amount of events before waking up
+    pub fn wakeup_events(self, wakeup_events: u32) -> PerfBufferBuilder<'a, 'b> {
```

Comment:
Seems inconsistent to accept `u32` here, given that the `buffer_buffer` API you add next accepts a `usize` index and then casts to `i32`, no?

Reply:
The reason it's u32 is because that's the type in libbpf
```
union {
    __u32       wakeup_events;    /* wakeup every n events */
    __u32       wakeup_watermark; /* bytes before wakeup   */
};
```

Reply:
I understand. But it's also an implementation detail. It's obviously minor, but what I was trying to point out is that `buf_idx` as accepted [here](https://github.com/libbpf/libbpf-rs/pull/348/files#diff-753b0a913ef36046b385a6709fa7541513a1ace7a6bfa176fc55aae9c2c02c1aR228) is `usize` but it is cast to a `i32` for the call to `perf_buffer__buffer` as that is the internally used type. So if `libbpf` usage is the argument being made, why do it differently there but not here? Just trying to understand what rationale you apply.

I think either usage is a trade-off ultimately (one makes it harder to call while the other potentially gets us into trouble when casting) and so intuitively I'd treat both approaches as equally bad and find a third argument as a tiebreaker: in this case consistency. We expose `pages` as `usize` and then cast internally so it may be best to do the same here? Let me know what you think.

Reply:
But given that you removed the code in question I think we can resolve this issue :-)